students/ahwall/Lesson05Homework/MailRoomPt3.py

```diff
@@ -0,0 +1,151 @@
+try again\n")
+                            thank_you()
+                        else:
+                            vallist.append(DonationVal)
+                            print()
+                            print(ty_letter, "\b's donation has been added to \
+the Donor Report...Creating 'Thank you email'")
+                            i = 1
+                            One_letter(ty_letter, DonationVal)
+                            mainloop()
+                            break
+                    else:
+                        j += 1
+            else:  # WORKS
+                try:
+                    DonationVal = input("\nAdding Donor to list,\n\nHow much was \
+their donate?  ")
+                #  place exception to make sure int is given
+                    DonationVal = int(DonationVal)
+                    vallist = []
+                    vallist.append(DonationVal)
+                except ValueError:
+                    print("\nYou did not enter an numberic value, Please try \
+again\n")
+                    thank_you()
+                else:
+                    Don_Val = ()
+                    Don_Val = (ty_letter[:], vallist[:])
+                    donors.append(Don_Val)
+                    print()
+                    print(ty_letter, "and their donation have been added to \
+                    the Donor Report...Creating 'Thank you email'")
+                    i = 1
+                    One_letter(ty_letter, DonationVal)
+                    mainloop()
+                break  # Works
+
+
+def One_letter(donor, donation):
+    print()
+    print("Dear {};\n\n\tThank you for your generous donation of ${:.2f}.\n\n \
+\tSincerely,\n\n \tThe Python Project\n".format(donor, donation))  # Works
+
+
+def All_letters(donordict):
+    donorvallist = list(donordict.values())
+    for i in range(len(donorvallist[0])):
+        Donor, Donation = donordict['Donor'][i], donordict['Donation'][i]
+        Donation = int(Donation)
+        # could not figure out a way to unpack the dictionary (like Mr. Barker
+        # suggested in the question) with multiple values per key, since you
+        # can not use [i] because it needs to be a int not a string
+        lettertemplate = ("Dear {},\n\n\
+        Thank you for your very kind donation of ${:,.02f}. \
+        \n\n\t\t Sincerely,\n\t\t  -The Team".format(Donor, Donation))
+        filename = open("{}.txt".format(Donor), 'w')
+        filename.write(lettertemplate)
+        filename.close()
+    print("Creating 'Thank you' letters to all Donors")
+    mainloop()
+
+
+def report(donors):
+    Repheader = ('Donor Name', 'Total Given', 'Num Gifts', 'Average Gifts')
+    headerspace = '{:20}| {:^14}| {:^10}| {:^14}'
+    ReportCats = headerspace.format(*Repheader)
+    print('\n', ReportCats)
+    print('-' * 63)
+    # for loop to print report
+    [print('{:20}| ${:^13,.02f}| {:^10}| ${:^14,.02f}'.format(item[0],
```

Comment:
Nice list comprehension!  I think the ideal case would put the print separate from the list comp, so you only have to call print() once, but this works!